rust/Earthfile

```diff
@@ -1,54 +1,111 @@
+        for source_lib in \$source_libs; do
+          find . -maxdepth 1 -regex \"\./\$source_lib-[^-]+\" -exec bash -c 'readlink -f {}; rm -rf {}' \; ;
+        done
+      done"
+
+# get-tomljson gets the portable tomljson binary.
+get-tomljson:
+    FROM alpine:3.18.3
+    ARG USERARCH
+    ARG version=2.1.0
+    RUN wget -O tomljson.tar.xz https://github.com/pelletier/go-toml/releases/download/v${version}/tomljson_${version}_linux_${USERARCH}.tar.xz; \
+        tar -xf tomljson.tar.xz; \
+        chmod +x tomljson
+    SAVE ARTIFACT tomljson
 
-# get-stoml gets the portable stoml binary
-get-stoml:
+# get-jq gets the portable jq binary.
+get-jq:
     FROM alpine:3.18.3
-    RUN wget -O stoml https://github.com/freshautomations/stoml/releases/download/v0.7.1/stoml_linux_amd64; \
-       chmod +x stoml
-    SAVE ARTIFACT ./stoml stoml
+    ARG USERARCH
+    ARG version=1.7
+    RUN wget -O jq https://github.com/jqlang/jq/releases/download/jq-${version}/jq-linux-${USERARCH}; \
+        chmod +x jq
+    SAVE ARTIFACT jq
+
+# INSTALL_CARGO_SWEEP installs cargo-sweep if it doesn't exist already.
+INSTALL_CARGO_SWEEP:
+    COMMAND
+    DO +RUN_WITH_CACHE --command="set -e;
+        if [ ! -f \$CARGO_HOME/bin/cargo-sweep ]; then
+          cargo install cargo-sweep --root \$CARGO_HOME;
+        fi;"
+
+# RUN_WITH_CACHE runs the passed command with the CARGO caches mounted.
+# Arguments:
+#   - command (required): Command to run, can be any expression.
+#
+# This implementation is not expected to significantly change. Prefer using the `CARGO` UDC if you can, so you can get future improvements transparently.
+RUN_WITH_CACHE:
+    COMMAND
+    ARG command
+    ARG EARTHLY_TARGET_PROJECT_NO_TAG
+    ARG CACHE_ID="${EARTHLY_TARGET_PROJECT_NO_TAG}#earthly-cargo-cache"
+    # $ORIGINAL_CARGO_HOME/bin will contain the cargo and rust binaries, as well as the installed crates.
+    # We save it to reset it back at the end. This location is presumed to be in the calling target layers filesystem rather than in other mount cache.
+    ARG ORIGINAL_CARGO_HOME=$CARGO_HOME
+    ARG ORIGINAL_CARGO_INSTALL_ROOT=$CARGO_INSTALL_ROOT
+    # Make sure that crates installed though this UDC are stored in the original cargo home, and not in the cargo home within the mount cache.
+    # This way, if BK garbage-collects them, the build is not broken
+    ENV CARGO_INSTALL_ROOT=$ORIGINAL_CARGO_HOME
+    # We need $CARGO_HOME/.package-cache within the earthly shared-mount-cache, so cargo can properly synchronize parallel access to $CARGO_HOME resources.
+    ENV CARGO_HOME="/earthly/.cargo"
+    RUN echo $CACHE_ID
+    RUN --mount=type=cache,mode=0777,id=$CACHE_ID,sharing=shared,target=/earthly \
+        --mount=type=cache,mode=0777,target=target \
+    set -e; \
+    mkdir -p $CARGO_HOME; \
+    printf "Running:\n      $command\n"; \
+    eval $command
+    ENV CARGO_HOME=$ORIGINAL_CARGO_HOME
+    ENV CARGO_INSTALL_ROOT=$ORIGINAL_CARGO_INSTALL_ROOT
```

Comment:
newline

Reply:
we should add a newline linter such as the one from earthly's root Earthfile for this.